embedded/tbtree/reader.go

```diff
@@ -230,18 +230,18 @@ func (r *Reader) Read() (key []byte, value []byte, ts, hc uint64, err error) {
 			}
 		}
 
-		if len(r.prefix) == 0 {
-			return cp(leafValue.key), cp(leafValue.value), leafValue.ts, leafValue.hCount, nil
+		// prefix mismatch
+		if len(r.prefix) > 0 &&
+			(len(leafValue.key) < len(r.prefix) || !bytes.Equal(r.prefix, leafValue.key[:len(r.prefix)])) {
+			continue
 		}
 
-		if len(r.prefix) > 0 && len(leafValue.key) >= len(r.prefix) {
-			leafPrefix := leafValue.key[:len(r.prefix)]
-
-			// prefix match
-			if bytes.Equal(r.prefix, leafPrefix) {
-				return cp(leafValue.key), cp(leafValue.value), leafValue.ts, leafValue.hCount, nil
-			}
+		if r.skipped < r.offset {
+			r.skipped++
```

Comment:
How does it work with entries marked as deleted or the expired ones?

Reply:
Are we still introducing skipping entries on the btree level ? I think that currently there's no valid use case for that thus it would only be a technical depth ATM.